homework/index.js

```diff
@@ -30,13 +32,126 @@
+        if (value !== null) {
+          if (key === 'name') {
+            const tableRow = createAndAppend('tr', parentTable, { class: 'tr' });
+            createAndAppend('td', tableRow, {
+              text: 'repository :',
+              class: 'title',
+            });
+            const tdName = createAndAppend('td', tableRow);
+            createAndAppend('a', tdName, { text: value, href: obj.repoUrl, target: '_blank' });
+          }
+          if (key === 'description' || key === 'forks') {
+            const tableRow = createAndAppend('tr', parentTable, { class: 'tr' });
+            createAndAppend('td', tableRow, {
+              text: `${key} :`,
+              class: 'title',
+            });
+            createAndAppend('td', tableRow, { text: value });
+          }
+          if (key === 'updated') {
+            const tableRow = createAndAppend('tr', parentTable, { class: 'tr' });
+            createAndAppend('td', tableRow, {
+              text: 'updated :',
+              class: 'title',
+            });
+            createAndAppend('td', tableRow, {
+              text: `${value.slice(0, 10)}, ${value.slice(11, 19)}`,
+            });
+          }
+        }
+      });
+    } else {
+      parentTable.innerHTML = '';
+      showRepoDetailsInTable(obj, parentTable);
+    }
+  }
+
+  function showContributorsAsList(obj, parent) {
+    fetchJSON(obj.contributorsUrl, (error, data) => {
+      if (error) {
+        createAndAppend('div', parent, { text: error.message, class: 'alert-error' });
+      } else {
+        parent.innerHTML = '';
+        if (data === null) {
+          createAndAppend('div', parent, { text: 'No Contribution Data.', class: 'alert-error' });
+        } else {
+          createAndAppend('p', parent, { text: 'Contributions', class: 'contributors-header' });
+          const ul = createAndAppend('ul', parent, { class: 'contributors-list' });
+
+          data.forEach(contributor => {
+            const li = createAndAppend('li', ul, { class: 'contributor-item' });
+            createAndAppend('img', li, {
+              src: contributor.avatar_url,
+              alt: contributor.login,
+              class: 'contributor-avatar',
+            });
+            const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+            createAndAppend('div', contributorData, { text: contributor.login });
+            createAndAppend('div', contributorData, {
+              text: contributor.contributions,
+              class: 'contribution-count',
+            });
+            li.addEventListener('click', () => {
+              window.open(contributor.html_url);
+            });
+          });
+        }
+      }
+    });
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
```

Comment:
In case of an error it is best to 'exit early' and leave out an `else`.  In the rest of the code you can now do without the extra indentation, which makes the code more readable.

```js
  fetchJSON(url, (err, data) => {
    const root = document.getElementById('root');
    if (err) {
      createAndAppend('div', root, { text: err.message, class: 'alert-error' });
      return;
    }
   const header = createAndAppend('header', root, { class: 'header' });
   ...
```